### ISSUE TITLE
Setting a variable named mode breaks some systemctl completions in zsh

### ISSUE THREAD

**Issue by unquote7083** [systemd/systemd#24473]
### systemd version the issue has been seen with

251

### Used distribution

arch

### Linux kernel version used

5.19.3

### Component

systemctl

### Unexpected behaviour you saw

A variable named `mode` in zsh  breaks completions like `systemctl status`

Looks to be caused by `$mode` being used in 

https://github.com/systemd/systemd/blob/6dd3b818bbfc6a2c35e5a530186a8c0bc1e01118/shell-completion/zsh/_systemctl.in#L195-L196

it's also used in a couple other places in that file.

### Steps to reproduce the problem

```
$ zsh --norcs
$ autoload -Uz compinit; zstyle \* menu select; compinit

# completions work fine
$ systemctl status <tab>

$ mode=foo

# completions are broken 
$ systemctl status <tab>
```

**Comment by bluca** [comment]
None of us uses zsh, please send a PR with a fix

**Comment by slmb3** [comment]
I couldn't understand how `mode` variable is supposed to be set

**Comment by yuwata** [comment]
Fix is waiting in #24476.

> I couldn't understand how `mode` variable is supposed to be set

I guess that's mistakenly imported from bash completion for systemctl.
